Connectors/Navisworks/Speckle.Connectors.NavisworksShared/Operations/Send/Filters/NavisworksSavedViewsFilter.cs

```diff
@@ -0,0 +1,173 @@
+
+public class NavisworksSavedViewsFilter : DiscriminatedObject, ISendFilterSelect
+{
+  private readonly IElementSelectionService _selectionService;
+
+  public NavisworksSavedViewsFilter(IElementSelectionService selectionService)
+  {
+    _selectionService = selectionService;
+
+    Items = [];
+    SelectedItems = [];
+
+    GetSavedViews();
+  }
+
+  public string Id { get; set; } = "navisworksSavedViews";
+
+  public string Name { get; set; } = "Saved Views";
+
+  public string Type { get; set; } = "Select";
+
+  public string? Summary { get; set; }
+
+  public bool IsDefault { get; set; }
+
+  public List<string> SelectedObjectIds { get; set; } = [];
+
+  public Dictionary<string, string>? IdMap { get; set; }
+
+  public bool IsMultiSelectable { get; set; }
+
+  public List<SendFilterSelectItem> SelectedItems { get; set; }
+
+  public List<SendFilterSelectItem> Items { get; set; }
+
+  public List<string> RefreshObjectIds()
+  {
+    List<string> objectIds = [];
+
+    if (SelectedItems.Count == 0)
+    {
+      return objectIds;
+    }
+
+    var savedViews = NavisworksApp.ActiveDocument.SavedViewpoints;
+
+    foreach (var savedViewItem in SelectedItems.Select(item => ResolveSavedView(item.Id)))
+    {
+      // Get the visible elements in the saved view.
+      objectIds.AddRange(ResolvedSavedViewObjects(savedViewItem));
+    }
+
+    return objectIds;
+  }
+
+  private static NAV.SavedViewpoint ResolveSavedView(string savedViewReference)
+  {
+    if (Guid.TryParse(savedViewReference, out var guid))
+    {
+      // Even though we may have already got a match, that could be to a generic Guid from earlier versions of Navisworks
+      if (savedViewReference != Guid.Empty.ToString())
+      {
+        return (NAV.SavedViewpoint)NavisworksApp.ActiveDocument.SavedViewpoints.ResolveGuid(guid);
+      }
+    }
+
+    var savedRef = new NAV.SavedItemReference("LcOpSavedViewsElement", savedViewReference);
+
+    var resolvedReference = NavisworksApp.ActiveDocument.ResolveReference(savedRef) as NAV.SavedViewpoint;
+
+    return resolvedReference
+      ?? throw new SpeckleSendFilterException($"Saved view with reference {savedViewReference} not found.");
+  }
+
+  private IEnumerable<string> ResolvedSavedViewObjects(NAV.SavedViewpoint savedView)
+  {
+    var objectIds = new List<string>();
+
+    // THIS IS COMMENTED OUT AS IT IS LEGACY DEFENSIVE BEHAVIOUR - DISCUSSION REQUIRED
+    // if (!savedView.ContainsVisibilityOverrides)
+    // {
+    //   // We check this again as the view settings may have changed in the saved card.
+    //   // If the saved view does not contain visibility overrides, this is effectively everything in the model.
+    //   // This will need to be the documented behaviour.
+    //   throw new SpeckleSendFilterException(
+    //     "Saved view does not contain visibility overrides. This would effectively publish everything in the model."
+    //   );
+    // }
```

Comment:
Willing to discuss either uncommenting or removing this. An opinion expressed in v2 code was that this was valuable, partly because sends were slower then, but also because this class of viewpoint doesn't hide anything from view; it's no different from sending everything.

This PR keeps the legacy position but commented out.

Reply:
happy to leave it, no idea if it's correct or not